### PR TITLE
Update OutOfMemoryException message

### PR DESCRIPTION
**Pull request by eduardo-vp**
Fixes #71761 and #55964

### CODE REVIEW

File: src/coreclr/inc/ex.h [hunk 1]
```diff
@@ -573,7 +573,7 @@ class OutOfMemoryException : public Exception
     }
 
     HRESULT GetHR() {LIMITED_METHOD_DAC_CONTRACT;  return E_OUTOFMEMORY; }
-    void GetMessage(SString &result) { WRAPPER_NO_CONTRACT; result.SetASCII("Out Of Memory"); }
+    void GetMessage(SString &result) { WRAPPER_NO_CONTRACT; result.SetASCII("Out of memory or another resource (e.g. Threads, PIDs)."); }
```
Comment (danmoseley) [inline src/coreclr/inc/ex.h:576]:
```suggestion
    void GetMessage(SString &result) { WRAPPER_NO_CONTRACT; result.SetASCII("Out of memory or reached a resource limit such as of threads or processes."); }
```
perhaps prettier

**Comment by danmoseley** [comment]
@jkotas any position on the message?

**Comment by jkotas** [comment]
If there are places where we are throwing OutOfMemoryException in situations that are not actually out-of-memory, we should fix those places. We have done that already for multiple of them. For example, look for: `Internal limitation: attempt to create more than 2^26 SyncBlocks.` or `Insufficient memory within specified address space range to continue the execution of the program.`.

I do not think we should be modifying the default message for OutOfMemoryException. It does not help with diagnosing the problems that are not actual out of memory exceptions.

**Comment by danmoseley** [comment]
Yea even better. Customer messages, but same type.
Here's a link to one of those mentioned

https://github.com/dotnet/runtime/blob/2201016c1e13bdb9abf49e2e38cadf4ee0568df2/src/coreclr/vm/syncblk.cpp#L843

@eduardo-vp You may need to reproduce a particular case locally in order to confidently fix its message.

**Pull request closed without merging**